crates/nu-data/src/base.rs

```diff
@@ -64,20 +64,34 @@ pub fn select_fields(obj: &Value, fields: &[String], tag: impl Into<Tag>) -> Val
     out.into_value()
 }
 
-pub fn reject_fields(obj: &Value, fields: &[String], tag: impl Into<Tag>) -> Value {
+pub fn reject_fields(
+    obj: &Value,
+    fields: &[Tagged<String>],
+    tag: impl Into<Tag>,
+) -> Result<Value, ShellError> {
     let mut out = TaggedDictBuilder::new(tag);
 
     let descs = obj.data_descriptors();
 
+    for field in fields {
+        if descs.iter().all(|desc| desc != &field.item) {
+            return Err(ShellError::labeled_error(
+                "Invalid column",
+                "invalid column",
+                field.tag.clone(),
+            ));
+        }
+    }
+
```

Comment:
The original for loop below this does not give access to each field, which we need to grab the tag for the error, so I had to basically those original nested loops and flip them inside out in order to get the check to work.

Reply:
I was wondering about that. we want to check that at least one of the rows has a column with that name, right? If some don't but others do, seems like that'd still be okay.

Reply:
Hmm, where do those situations come up?

```shell
> open Cargo.toml | get bin | flatten required-features | last 3
───┬─────────────────────────────┬────────────────────────────────────────────┬───────────────────
 # │            name             │                    path                    │ required-features 
───┼─────────────────────────────┼────────────────────────────────────────────┼───────────────────
 0 │ nu_plugin_extra_from_sqlite │ src/plugins/nu_plugin_extra_from_sqlite.rs │ sqlite            
 1 │ nu_plugin_extra_to_sqlite   │ src/plugins/nu_plugin_extra_to_sqlite.rs   │ sqlite            
───┴─────────────────────────────┴────────────────────────────────────────────┴───────────────────
───┬──────┬─────────────
 # │ name │    path     
───┼──────┼─────────────
 2 │ nu   │ src/main.rs 
───┴──────┴─────────────
> open Cargo.toml | get bin | flatten required-features | last 3 | reject required-features
───┬─────────────────────────────┬────────────────────────────────────────────
 # │            name             │                    path                    
───┼─────────────────────────────┼────────────────────────────────────────────
 0 │ nu_plugin_extra_from_sqlite │ src/plugins/nu_plugin_extra_from_sqlite.rs 
 1 │ nu_plugin_extra_to_sqlite   │ src/plugins/nu_plugin_extra_to_sqlite.rs   
───┴─────────────────────────────┴────────────────────────────────────────────
error: Invalid column
  ┌─ shell:1:73
  │
1 │ open Cargo.toml | get bin | flatten required-features | last 3 | reject required-features
  │                                                                         ^^^^^^^^^^^^^^^^^ invalid column 
```

Reply:
I guess I'm not really sure how to deal with that situation.

Reply:
Yeah exactly. When you have a mix of rows that have different shapes. For these I think we should detect if we successfully removed the column at least once. If we get to the end and never did, we should give the error then.

Reply:
Would that imply that reject and select would no longer be able to stream their results?  As we would have to evaluate all the rows in order to know that at least one row has the column?

Reply:
They can still stream. It's just that at the end we'd also send an error. For some cases the user would never know, like:

```
ls | select nammme
```

Would give no results and error at the end of the stream. If there's no end then there still *may* be a row that matches, so we keep looking. Most people aren't doing infinite streams, so they should fall into the happy case.

Reply:
Sorry for the delay here, I do plan to figure out a way to do this, will probably do it over the weekend.